lib/rpmsg/rpmsg.c

```diff
@@ -227,7 +227,7 @@ int rpmsg_create_ept(struct rpmsg_endpoint *ept, struct rpmsg_device *rdev,
 	rpmsg_init_ept(ept, name, addr, dest, cb, unbind_cb);
 	rpmsg_register_endpoint(rdev, ept);
 
-	if (rdev->support_ns && ept->dest_addr == RPMSG_ADDR_ANY) {
+	if (ept->name[0] && rdev->support_ns && ept->dest_addr == RPMSG_ADDR_ANY) {
```

Comment:
why there is endpoint without a name but it doesn't have dest_addr? I had thought the endpoint without a name is static endpoint.

Reply:
Yes, the static endpoint should always have the dest_addr, the modification just make it's symmetry with rpmsg_create_ept

Reply:
name is empty and dest address is ANY endpoint is just invalid, it looks like this should be check when creating the endpoint, and then, we don't need to check if the name is empty or not here.

Reply:
Yes, it's better to avoid the invalid combination in the first place.